pkg/project/project.go

```diff
@@ -82,26 +87,81 @@ func Exists(client *occlient.Client, projectName string) (bool, error) {
 	return true, nil
 }
 
+// GetMachineReadableFormat gathers the readable format and output a Project struct
+// for json to marshal
 func GetMachineReadableFormat(projectName string, isActive bool, apps []string) Project {
-
 	return Project{
 		TypeMeta: metav1.TypeMeta{
 			Kind:       "Project",
 			APIVersion: "odo.openshift.io/v1alpha1",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name: projectName,
+			Name:      projectName,
+			Namespace: projectName,
 		},
 		Spec: ProjectSpec{
 			Applications: apps,
 		},
 		Status: ProjectStatus{
-
 			Active: isActive,
 		},
 	}
 }
 
+// MachineReadableSuccessOutput ...
+func MachineReadableSuccessOutput(projectName string, message string) {
+	machineOutput := machineoutput.Success{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Project",
+			APIVersion: "odo.openshift.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      projectName,
+			Namespace: projectName,
+		},
+		Message: message,
+	}
+
+	printableOutput, err := json.Marshal(machineOutput)
+
+	// If we error out... there's no way to output it (since we disable logging when using -o json)
+	if err != nil {
+		fmt.Printf("Unable to unmarshal JSON: %s", err.Error())
+		os.Exit(1)
```

Comment:
Exit code for failure to create a project and for failure to generate the machine readable output is same. How would someone who is using the API determine what the exact cause of error?

Reply:
Ideally this should never happen, but unfortunately we do hide all non-json related output. What would happen upstream is that VSCode will be unable to marshal the JSON output anyways (it will output "Unable to unmarshal json... from Odo..").

I honestly couldn't think of a better solution.

Reply:
Yes, I kind of assumed that the message (in this case `"Unable to unmarshal JSON`) would be a differentiator, but it still feels a bit skewed. I might sound pedantic here but if we have more such ambiguities w.r.t exit codes, it might come back to bite us since our APIs are going to be crucial for projects like Adapters. Correct me if I'm wrong.